writer_test.go

```diff
@@ -565,3 +570,28 @@ func TestBloomFilterForDict(t *testing.T) {
 		t.Error("bloom filter should have contained 'test'")
 	}
 }
+
+func TestEmptyFile(t *testing.T) {
+	tmp, err := os.CreateTemp("/tmp", "*.parquet")
+	if err != nil {
+		t.Error(err)
+	}
+	defer tmp.Close()
+	path := tmp.Name()
+	defer os.Remove(path)
+
+	writer := parquet.NewWriter(tmp)
+	if err := writer.Close(); err != nil {
+		t.Error(err)
+	}
+
+	fs, err := tmp.Stat()
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = parquet.OpenFile(tmp, fs.Size())
+	if !errors.Is(err, parquet.ErrMissingRootColumn) {
+		t.Errorf("expected error %s, got %s", parquet.ErrMissingRootColumn, err)
+	}
+}
```

Comment:
In this test there is no schema defined because none were passed in when constructing the writer, and no rows were written so it assumes an empty schema.

Could we add a test where the writer is given a schema as well?

Reply:
Added a test. Working nicely and creating valid empty file.